src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -214,7 +222,7 @@ private async Task ProcessQueueAsync()
                         // Non mutating are fire-and-forget because they are by definition readonly. Any errors
                         // will be sent back to the client but we can still capture errors in queue processing
                         // via NFW, though these errors don't put us into a bad state as far as the rest of the queue goes.
-                        _ = work.CallbackAsync(context, cancellationToken).ReportNonFatalErrorAsync();
+                        _ = work.CallbackAsync(context, cancellationTokenSource.Token).SafeContinueWith(t => cancellationTokenSource.Dispose(), TaskScheduler.Default).ReportNonFatalErrorAsync();
```

Comment:
i think it would be cleaner to extract the try/await/finally into a helper async method.  then either await it above, or fire/forget it here.  that way, the disposal logic is just in that single place.

Reply:
Good idea, thanks. I thought about just putting it in the callback, but that felt weird so I didn't, but obviously stopped thinking :)